### PR TITLE
[stable/spinnaker] Creating storageClassName and storage volume variable

### PR DESCRIPTION
**Pull request by ElizabethSvit**
#### What this PR does / why we need it:

One might need to be able to pass a custom storageClass name and a different storage volume. PR does provide an additional field in values.yaml for volumeClaim (storageClassName and storage).

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by rochaporto** [comment]
Thanks @ElizabethSvit we bumped into the same need. Patch seems to work locally.

File: stable/spinnaker/values.yaml [hunk 1]
```diff
@@ -229,3 +229,7 @@ serviceAccount:
   # If left blank it is auto-generated from the fullname of the release
   halyardName:
   spinnakerName:
+
+volumeClaim:
+  storageClassName:
```
Comment (dioguerra) [inline stable/spinnaker/values.yaml]: Here, as this is not mandatory, should be commented like so:
# storageClassName: default

File: stable/spinnaker/templates/statefulsets/halyard.yaml [hunk 2]
```diff
@@ -149,7 +149,8 @@ spec:
       labels:
 {{ include "spinnaker.standard-labels-base" . | indent 8 }}
     spec:
+      storageClassName: {{ .Values.volumeClaim.storageClassName }}
```
Comment (dioguerra) [inline stable/spinnaker/templates/statefulsets/halyard.yaml]: This should be wrapped under an if condition.

If the storageClassName is not defined the PV should default to the k8s defined default SC name.
Instead, what you are doing is defining storageClassName to blank. If i use the template with the default values.yaml wont work?

Can you confirm?

Comment (ElizabethSvit) [reply stable/spinnaker/templates/statefulsets/halyard.yaml]: Yes, you are right, a space value as a default won't work so I put an if statement to check if the field "storageClassName" is defined in values.yaml first.

**Comment by MrD2** [comment]
/assign dwardu89

Can we have this merged?

**Comment by ElizabethSvit** [comment]
/assign paulczar 
Can you briefly look through this PR?

**Comment by rochaporto** [comment]
Anything missing to get this merged in?

**Comment by zanhsieh** [comment]
@ElizabethSvit 
Would you mind to resolve the conflict of files please? Thank you.
stable/spinnaker/Chart.yaml
stable/spinnaker/templates/statefulsets/halyard.yaml
stable/spinnaker/values.yaml

**Pull request closed without merging**